gms/gossiper.cc

```diff
@@ -902,7 +902,8 @@ void gossiper::run() {
                     logger.debug("Set live nodes to talk: endpoint_state_map={}, all_live_nodes={}, endpoints_to_talk_with={}",
                             _endpoint_state_map.size(), _live_endpoints, _endpoints_to_talk_with);
                 }
-                if (_endpoints_to_talk_with.empty()) {
+                if (_endpoints_to_talk_with.empty() ||
+                        _live_endpoints.size() < get_token_metadata_ptr()->get_all_endpoints().size()) {
```

Comment:
This will end up with talking to a the seed node configured only. For example, 10 nodes in the cluster, n1 to n10, n10 is down. Assume the seed node is an old ip (X) which does not point to any node in the cluster anymore. With this patch, the live_endpoints is 9, get_all_endpoints is 10. The branch you added will trigger, X will be added to the _endpoints_to_talk_with. 

The code below pop nodes from _endpoints_to_talk_witth e.g, {n1, n2, ... n9, X}, say n1, to talk with. 

Next round,  the branch added will trigger again, which ends up like  {n2, ... n9, X, X}.

Reply:
Ugh. This is certainly not the intent. My goal is that do it only once in a while, wwithout giving this node any preference. How do I accomplish that?

Reply:
Please read the scenario described in the commit message. The node can figure out its host id and old and IP address, but it can't figure out its new IP address or new IP addresses of its peers before it contacts them - they are in the seed list and not associated with any host id.

Reply:
I don't understand what you suggest specifically. I updated the patch to address your comment re accumulating seeds in endpoints_to_talk_with. Please take a look or be more specific, what you suggest to change and where.